.github/workflows/saas-preview-environment.yaml

```diff
@@ -0,0 +1,38 @@
+name: SaaS preview environment
+on:
+  pull_request:
+    types:
+      - synchronize
+      - labeled
+      - unlabeled
+      - opened
+      - closed
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  create_preview_environment:
+    name: Create or migrate preview environment
+    if: ${{ github.event.action != 'closed' && contains(github.event.pull_request.labels.*.name, 'preview-environment') }}
+    runs-on: runs-on=${{ github.run_id }}/runner=sw-amd64
+    steps:
+      - name: Trigger preview environment
+        uses: shopware/saas/.github/actions/preview-environment@trunk
```

Comment:
Does this even work from public repos? I think we need add a check for premissions